app/assets/javascripts/discourse/controllers/topic.js.es6

```diff
@@ -905,6 +905,39 @@ export default Ember.Controller.extend(bufferedProperty("model"), {
       }
     },
 
+    joinGroup() {
+      const groupId = this.get("model.group.id");
+      if (groupId) {
+        if (this.get("model.group.allow_membership_requests")) {
+          const groupName = this.get("model.group.name");
+          return ajax(`/groups/${groupName}/request_membership`, {
+            type: "POST",
+            data: {
+              reason: I18n.t("topic.group_request_reason", {
```

Comment:
It's a little odd to me to have the client insert the reason like this. I would assume as reason parameter would be designed for the user to type in a custom reason.

I often think of a 3rd party consumer of our API. Would we expect someone using the API gem to create this reason the same way?

Perhaps we instead pass the topic_id and no reason, and have the server build the string?